library/block/block_extensions/src/main/java/org/quiltmc/qsl/block/extensions/mixin/MixinBlock.java

```diff
@@ -0,0 +1,88 @@
+package org.quiltmc.qsl.block.extensions.mixin;
+
+import net.minecraft.block.AbstractBlock;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.state.StateManager;
+import net.minecraft.state.property.Property;
+import org.apache.commons.compress.utils.Lists;
+import org.quiltmc.qsl.block.extensions.impl.BlockExtension;
+import org.quiltmc.qsl.block.extensions.impl.QuiltBlockImpl;
+import org.slf4j.Logger;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+
+import java.util.LinkedList;
+
+@Mixin(Block.class)
+public abstract class MixinBlock implements BlockExtension {
+
+    @Shadow protected abstract Block asBlock();
+    @Shadow @Final private static Logger LOGGER;
+    private final @Unique LinkedList<Block> proxies = new LinkedList<>();
+
+    @Inject(method = "<init>",
+            at = @At(value = "INVOKE", target = "Lnet/minecraft/state/StateManager$Builder;build(Ljava/util/function/Function;Lnet/minecraft/state/StateManager$Factory;)Lnet/minecraft/state/StateManager;"),
+            locals = LocalCapture.CAPTURE_FAILHARD)
+    private void injectProxyProperties(AbstractBlock.Settings settings, CallbackInfo ci, StateManager.Builder<Block, BlockState> builder) {
+        if (QuiltBlockImpl.PROXY_BLOCKS_TEMP_CONTAINER.get() != null && QuiltBlockImpl.PROXY_BLOCKS_TEMP_CONTAINER.get().length > 0){
+            var properties = Lists.newArrayList();
```

Comment:
```suggestion
            var properties = new ArrayList<Property>();
```

Please don't use stuff reminiscent from the before diamond generic Java versions.